cty/convert/conversion_collection.go

```diff
@@ -15,18 +15,18 @@ func conversionCollectionToList(ety cty.Type, conv conversion) conversion {
 	return func(val cty.Value, path cty.Path) (cty.Value, error) {
 		elems := make([]cty.Value, 0, val.LengthInt())
 		i := int64(0)
-		path = append(path, nil)
+		elemPath := append(path.Copy(), nil)
```

Comment:
These changes throughout are for clarity, now that the `path` parameter can be used multiple times (including in the call to `conversionUnifyCollectionElements`). Always mutating a copy seemed clearer to me than repeatedly mutating the passed parameter.